### PR TITLE
Esc add member tweak

### PR DESCRIPTION
**Pull request by sam-miele**

### CODE REVIEW

**Review by thebailers — APPROVED** [review]

File: conf/messages [hunk 1]
```diff
@@ -148,8 +151,7 @@ undertaking.confirmation.p4=If you have any other businesses, you must add them
 
 businessEntityEori.eoriInUse.error=This EORI is in use.
 
-#account.title is temporary and can be changed later, as test were failing because of missing account.title
-account.title = {0}
+account-homepage.title = {0}
```
Comment (priyankadhiman3006) [inline conf/messages:154]: If you are changig this, then you need to change test case where I have used #account.title

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala [hunk 2]
```diff
@@ -312,13 +313,20 @@ class BusinessEntityController @Inject()(
...
-    "businessEntityEori.regex.error",
-    fields => fields match {
-      case a if a.value.matches(EORI.regex) => true
-      case _ => false
-    }
-  ))
+    mapping("businessEntityEori" -> mandatory("businessEntityEori"))(eoriEntered => FormValues(s"$eoriPrefix$eoriEntered"))(eori => eori.value.drop(2).some)
+    .verifying("businessEntityEori.error.incorrect-length",
+      fields => fields match {
+        case a if a.value.toString().length == 14 || a.value.toString().length == 17 => true
```
Comment (priyankadhiman3006) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:319]: out of curosity why we are checking for 14 and 17 , isn't the eoRI length should be 12 or 15 including the GB part?

Comment (sam-miele) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:319]: Excluding the GB part is what we understand.

Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:316]: but then   it contradicts this, isn't?

```
  type EORI = String @@ EORI.Tag
  object EORI extends RegexValidatedString(
    """^(GB|XI)[0-9]{12,15}$"""
```

**Review by priyankadhiman3006 — APPROVED** [review]

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala (hunk 2 above)
Comment (sam-miele) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:316]: Nahhh it's a bit tricky to read but that means...
"(GB or XI) followed by (a member of the alphabet 0-9 (syntax sugar for digits) 12 or 15 times)"

Comment (sam-miele) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:316]: E.e

`scala> """^(GB|XI)[0-9]{12,15}$""".r.matches("GB1234567890")
val res1: Boolean = false

scala> """^(GB|XI)[0-9]{12,15}$""".r.matches("GB123456789012")
val res2: Boolean = true

scala> """^(GB|XI)[0-9]{12,15}$""".r.matches("GB123456789012345")
val res3: Boolean = true`

Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:316]: yup got it as I was typing  🙈

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala [hunk 3]
```diff
@@ -309,13 +310,20 @@ class BusinessEntityController @Inject()(
...
+        case _ => false
+      }
+    )
+    .verifying("businessEntityEori.regex.error",
+      fields => fields match {
+        case a if a.value.matches(EORI.regex) => true
```
[the remark this answers is not in the document]
Comment (sam-miele) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala:322]: good point. not necessary. removing.

**Comment by carwynellis** [comment]
Ta for pushing the updates. LGTM.

**Pull request merged**